### ISSUE TITLE
Path to GitHub client

### ISSUE THREAD

**Issue by augustuswm** [oxidecomputer/progenitor#395]
Currently the `octorust` client is generated out of the `third-part-api-clients` repo against an outdated GitHub spec. Ideally progenitor could be used to generate this client against the current live spec instead. This is a tracking issue to track the list of known features needed to make this possible. See the individual tickets for more details.

For reference this list was compiled against: https://github.com/github/rest-api-description/blob/main/descriptions/api.github.com/api.github.com.json

```tasklist
- [ ] Add `text/x-markdown` as a supported request body type
- [ ] Handle enums that specify a default not present in the list of values
- [ ] #344 
- [ ] Handle enum values that contain meaningful non-alphanumeric characters
- [ ] Duplicate struct generation (needs to be validated)
- [ ] Recursive optional type generation (needs to be validated)
- [ ] #268 
- [ ] Support for servers parameter
- [ ] Solution for access / refresh token handling
```

**Comment by geoffreygarrett** [comment]
@augustuswm & @ahl, which of the above tasks relate to the following:

```bash
--- stdout
cargo:rerun-if-changed=./api.github.com.json
--- stderr
thread 'main' panicked at progenitor-github/build.rs:10:50:
called `Result::unwrap()` on an `Err` value: Error("data did not match any variant of untagged enum ReferenceOr", line: 65608, column: 3)
```
(?)
 
I could only find https://github.com/oxidecomputer/progenitor/issues/348. If desired my setup is as available in the zip in the PR demo https://github.com/oxidecomputer/progenitor/pull/857.

**Comment by ahl** [comment]
> Seems to be the lack of support for JSON schema types in OpenAPI deserialisation.

Yeah, that works in 3.1, but not in 3.0.x. Progenitor *will* support 3.1 at some point, but 1. there aren't that many 3.1 docs out there and 2. I have a stack of work I want to do in order to "propertly" support it

**Comment by geoffreygarrett** [comment]
@ahl I'm not sure if the project is open to PRs etc. or if development will continue, but there's this I was hoping might get a review https://github.com/oxidecomputer/progenitor/pull/857.

**Comment by geoffreygarrett** [comment]
🦗
